itou/openid_connect/pro_connect/views.py

```diff
@@ -0,0 +1,334 @@
+
+    # A token has been provided so it's time to fetch associated user infos
+    # because the token is only valid for 5 seconds.
+    # We don't really need to access user_info since all we need is already in the access_token.
+    # Should we remove this ?
+    user_data, error_redirection = _get_user_info(request, access_token)
+    if error_redirection:
+        return error_redirection
+
+    if "sub" not in user_data:
+        # 'sub' is the unique identifier from ProConnect, we need that to match a user later on.
+        return _redirect_to_login_page_on_error(error_msg="Sub parameter error.", request=request)
+
+    user_kind = pro_connect_state.data["user_kind"]
+    is_successful = True
+    pc_user_data = USER_DATA_CLASSES[user_kind].from_user_info(user_data)
+    pc_session_email = pro_connect_state.data.get("user_email")
+
+    if pc_session_email and pc_session_email.lower() != pc_user_data.email.lower():
+        if pro_connect_state.data["channel"] == ProConnectChannel.INVITATION:
+            error = (
+                "L’adresse e-mail que vous avez utilisée pour vous connecter avec ProConnect "
+                f"({pc_user_data.email}) "
+                f"ne correspond pas à l’adresse e-mail de l’invitation ({pc_session_email})."
+            )
+        else:
+            error = (
+                "L’adresse e-mail que vous avez utilisée pour vous connecter avec ProConnect "
+                f"({pc_user_data.email}) "
+                f"est différente de celle que vous avez indiquée précédemment ({pc_session_email})."
+            )
+        messages.error(request, error)
+        is_successful = False
+
+    user_created = False
+    try:
+        user, user_created = pc_user_data.create_or_update_user(is_login=pro_connect_state.data.get("is_login"))
+    except InvalidKindException:
+        existing_user = User.objects.get(email=user_data["email"])
+        _add_user_kind_error_message(request, existing_user, user_kind)
+        is_successful = False
+    except MultipleUsersFoundException as e:
+        # Here we have a user trying to update his email, but with an already existing email
+        # let him login, but display a message because we didn't update his email
+        messages.error(
+            request,
+            format_html(
+                "L'adresse e-mail que nous a transmis ProConnect est différente de celle qui est enregistrée "
+                "sur la plateforme des emplois, et est déjà associé à un autre compte. "
+                "Nous n'avons donc pas pu mettre à jour {} en {}. "
+                "Veuillez vous rapprocher du support pour débloquer la situation en suivant "
+                "<a href='{}'>ce lien</a>.",
+                e.users[0].email,
+                e.users[1].email,
+                global_constants.ITOU_HELP_CENTER_URL,
+            ),
+        )
+        user = e.users[0]
+
+    code_safir_pole_emploi = user_data.get("structure_pe")
```

Comment:
Normalement ProConnect est censé garder le même nom.  
Sauf que pour l'instant la connexion avec FT n'est pas encore en place donc je ne peux pas tester ce point là :roll_eyes: 
Je mets un commentaire just in case